src/test/java/test/dataprovider/issue2327/SampleTestCase.java

```diff
@@ -0,0 +1,32 @@
+package test.dataprovider.issue2327;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedList;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+@Listeners(CustomListener.class)
```

Comment:
You don't need this listener because all this listener is doing is printing stuff.